include/dsAudio.h

```diff
@@ -286,7 +286,7 @@ dsError_t  dsSetAudioCompression(intptr_t handle, int compression);
  * This function returns the dialog enhancement level of the audio port corresponding to the specified port handle.
  *
  * @param[in] handle - Handle for the output audio port
- * @param[out] level - Pointer to Dialog Enhancement level (Value ranges from 0 to 16)
+ * @param[out] level - Pointer to Dialog Enhancement level (Value ranges from 0 to 12)
```

Comment:
How are we planning to handle 2.4.1 IDK , 2.5 SDK versions here which has 0 to 16 levels?  We need to have these levels defined with respect to the SDK Version.  Please corret if I am wrong here.

Reply:
A agree.
The HAL API should be IDK agnostic and support any version underneath.
The approach we now take with new HAL APIs is to provide a capabilities function which exposes the min-max values allowed for the platform implementation.
Something like dsGetAudioCompressionCaps(out int * minCompression, out int * maxCompression).
In turn, the Thunder plug-in should similarly expose the caps up to the application to determine how to use it.

Reply:
@ddevadas53276 , is that something feasible to have the caps function here as well?

Reply:
My concern is the impact on existing usage that uses the range 0 to 16.
Changing from 0-16 to 0-12 will impact at a minimum Thunder interface.

The Thunder method 
[setMS12AudioCompression](https://rdkcentral.github.io/rdkservices/#/api/DisplaySettingsPlugin?id=setms12audiocompression)

Sets a range 0-10 for the scaling. If changed, will need to be performed in the Thunder implementation.

The use of:
dsGetAudioCompressionCaps(out int * minCompression, out int * maxCompression).

Will mean changes to all products. So very disruptive.
The question is whether we can avoid changing the MW.
The only solution is for the HAL to scale the input  from 0-16 to 0-12.
I do not like that as it hides fudge.
We could have a backward compatibility mode or if dsGetAudioCompressionCaps is NOT called then the range is 0-16 and scaled by the HAL. Very messy,

Any other ideas?
I'm not enamoured by any of the options.

Reply:
The Thunder method is [https://rdkcentral.github.io/rdkservices/#/api/DisplaySettingsPlugin?id=setdialogenhancement](https://rdkcentral.github.io/rdkservices/#/api/DisplaySettingsPlugin?id=setdialogenhancement)
It calls out a 0-16 range in the API documentation, so my expectation was that this would have to be changed to 0-12 for the new IDK.

That's why I proposed a new caps function to expose the min-max value here.

We also need to look at how the app layer uses these values.
As uses a hard coded value of 5/16 when enhance speech quality is enabled.
https://www.stb.bskyb.com/jira/browse/LLDEV-6779 

If this isn't adjusted, then in the new IDK it will be 5/12 which is about 10% higher!!  It should be knocked down to at least 4 to be closer to the original setting.

Do we know if the IDK provides linear rescaling of the old range to the and new?  Does it make sense to interpolate our old setting inside the new scale or have they just capped the upper range of the old IDK scale?

Doing the value remapping under-the-hood is never really a sustainable approach - a fudge as Pete says.

This would need sign off product we can't get close to the original subjective dialogue enhancement level.

Reply:
>We also need to look at how the app layer uses these values.
I did see [setMS12AudioCompression](https://rdkcentral.github.io/rdkservices/#/api/DisplaySettingsPlugin?id=setms12audiocompression)

It looks like apps set a range 0-10.

Reply:
The proposed change of just changing the header documentation can't be accepted.